operator/internal/handlers/lokistack_create_or_update.go

```diff
@@ -204,6 +204,20 @@ func CreateOrUpdateLokiStack(
 		}
 	}
 
+	proxyCfg, err := manifests.GetProxyConfig(k.Client)
+	if err != nil {
+		return err
+	}
+
+	// Create cluster proxy trusted CA bundle.
+	var lokiTrustBundle *corev1.ConfigMap
+	if proxyCfg != nil {
+		lokiTrustBundle, err = manifests.createOrGetTrustedCABundleConfigMap(manifests.LokiTrustedCAName, ctx, k, stack)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
As said above the Get part of this methods belongs to a dedicated handlers internal package. In addition we should not mix `create` with `get` in a single function here (as we do in other operators unfortunately) for a couple of reasons:
1. We don't rely in this operator on custom `createOrUpdate` functions as we use the generic one from the controller-runtime. This keep some sanity and uniformity in equality checks and create/update logic.
2. We keep `get` calls isolated in the handlers package and pass them down to the `manifests` builders via `manifests.Options`. This ensures that are our builders incl. conditional logic (e.g. openshift, feature flags) do not require active communication to the APIserver and remain easy to test via unit tests.

Besides the above the creation of this TrustedCABundle ConfigMap incl. the Volumes/VolumeMounts additions for the Loki Components is a conditional feature. This means we need an amendment in the LokiStack API for users to input/select the configMap that includes their TrustedCABundle. This enables making this feature k8s-native first. For OpenShift users can rely that we auto-populate the configmap via the dedicated inject-ca-bundle annotations. Latter should be dedicated patching code isolated in the `manifests.openshift` package.